### ISSUE TITLE
UTK - IIIF Print

### ISSUE THREAD

**Issue by jillpe** [notch8/utk-hyku#358]
# Summary

After completing #355, #354 test the full functionality of IIIF Print on UTK.

# Acceptance Criteria
- [ ] Verify the following is working, if not make follow-up tickets for them 
  - [ ] As a user, I can view parent and child images grouped together in a manifest
  - [ ] As a user, when I upload an import, PDFs are split into component pages, OCR'd and derivative files are created for each page via a configurable job
  - [ ] As a user, I can configure the derivative generation between the following options (JP2, PDF, txt, TIFF)
  - [ ] The gem is compatible with v2 and v3 manifest
  - [ ] The gem is compatible with dynamic metadata profiles
  - [ ] As a user, when I select a work type, it's child works show up in the viewer under the parent item
  - [ ] As a user, I can easily navigate between child and parent works in the Universal Viewer and they appear in a logical order
  - [ ] As a user, I can see all of the metadata properties of a parent and child work in the Universal Viewer
  - [ ] As a user, I can zoom in on the viewer
  - [ ] As a user, I can search the Universal Viewer for child or parent keywords
  - [ ] As a user, I can see my child or parent keyword searches highlighted on the show page
  - [ ] As a user, when I click on a viewer facet, it will redirect me to a search
  - [ ] As a user, I can add [/manifest.json] to the URL on a work show page to access its manifest in a JSON format

# Testing Instructions
**PDF Splitting**
The following should happen after creating a work and attaching a PDF to it via the UI, Bulkrax or CSV importer
- The PDF splits into childworks, with one PDF per page
- There should be a thumbnail associated with each childwork
- The childworks should be sorted via page number order
- The childworks should be the same model as the parent
- When you navigate into the childwork, json, txt, and xml formats should be available via the dropdown
  - [ ] UI
  - [ ] Bulkrax

**UV**
The following should happen when you open a file in the viewer
- Each page of the PDF is represented by an image and is sorted by page number order in the left-side menu
- You can navigate through the pages via the paginated navigation at the top of the viewer or by clicking on the page in the left-side menu
- You can zoom in 
- The collapsable right-side menu contains metadata for the parent and child that is currently displayed
- Faceted metadata in the collapsable right-side menu is linked to that faceted search
- You can search for parent/child metadata or text within the file via the text finder on the bottom part of the UV, and the text is highlighted where it appears
  - [ ] UI
  - [ ] Bulkrax

**Catalog Search**
A catalog search for parent/child metadata or text within the file results in ONLY the parent work being returned in the catalog search
    - [ ] UI
    - [ ] Bulkrax

# Notes

**Comment by jillpe** [comment]
# Testing Instructions

Upload a PDF work with relatively easy words to read on it

- [x] The PDF is split into component pages (shown at the bottom of the work page)
- [ ] ~Orignal file derivatives (accessed via the dropdown) are …~
- [ ] ~Child work derivatives (accessed via the dropdown) are…~
- [ ] ~I can download the original file and child works via the dropdown~
  - dropdown menus no longer present on work show page 

Open the work in the UV
- [x] The child works show up in the viewer in a logical order
- [x] I can zoom in in the Viewer
- [x] I can use the paginated navigation to navigate through the viewer

Search for parent or child keyword in the viewer
- [x] The word I searched is highlighted and the work it is associated with is flagged

Click the 2 arrows on the right side above ‘More Information’
- [x] I see all of the metadata properties for the parent and child work in the UV

Click on one of the faceable metadata properties in the ‘More Information’ Menu
- [x] I was redirected to a search under the facet I clicked on

Go back to the main search page
Search a parent or child keyword
- [x] The parent work displays in the search results 

Return to the work show page
Add `/manifest.json` to the end of the URL
- [x] You are taken to the works JSON formatted manifest

</details>

**Comment by jillpe** [comment]
~~Found a potential bug - please do not test yet~~ Okay to test now

**Comment by DiemBTran** [comment]
**Needs further review:** First attempt at QA did not pass.

tested on:
- https://dc.utk-hyku-staging.notch8.cloud/concern/books/029d2c63-1893-45a9-b8e0-114980a120ae?locale=en
- https://utk-hyku-staging.notch8.cloud/sidekiq/retries

# Testing instructions
- The following should happen after creating a work and attaching a PDF to it via the UI, Bulkrax or CSV importer
  - I created a book work with a PDF file through the UI 
- The PDF splits into childworks, with one PDF per page
  - Kirk and I checked Sidekiq to see that there were no more jobs in queue. We refreshed the work show page and did not see any child works split off from our PDF

<details><summary>No jobs in Sidekiq queue. Child works were not created for this book work with PDF</summary>

![Sidekiq 2023-04-12 at 9 25 17 AM](https://user-images.githubusercontent.com/29311858/231522569-218c73e3-64e7-4f37-bfcd-5801370c0f3e.jpg)
![Book 2023-04-12 #358 test IIIF print ID 029d2c63-1893-45a9-b8e0-114980a120ae Hyku 2023-04-12 at 9 25 07 AM (1)](https://user-images.githubusercontent.com/29311858/231522943-e275d678-9f8e-4795-ab34-a3d2bb8d9d1c.jpg)
</details>

**Comment by ShanaLMoore** [comment]
@DiemBTran after yesterday's client meeting I totally meant to pull this one back (or slap a blocked label on it) to do a better job at breaking down the testing instructions. Sorry! I dropped the ball big time.

**Comment by DiemBTran** [comment]
@ShanaLMoore no worries, tysm for the update ･ᴗ･

**Comment by ShanaLMoore** [comment]
:bug: UTK needs to be able to search all allinson flex metadata fields in the UV. This will be created as a separate ticket. https://github.com/scientist-softserv/utk-hyku/issues/460

ADL: https://github.com/scientist-softserv/adventist-dl/blob/main/app/controllers/catalog_controller.rb#L80-L81
UTK: https://github.com/scientist-softserv/utk-hyku/blob/main/app/controllers/catalog_controller.rb#L55-L59

**Comment by DiemBTran** [comment]
**This passes SoftServ QA:** See attached screenshot.

tested on:
- https://dc.utk-hyku-staging.notch8.cloud/concern/generic_works/9f20a65f-211e-49db-a995-5e4159716d84?locale=en

<details><summary>IIIF Print is working in UTK</summary>

![Generic Work 2023-07-12 🦄 Spike UTK - Test IIIF Print #358 ID 9f20a65f-211e-49db-a995-5e4159716d84 Hyku 2023-07-12 at 9 50 03 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/d468629c-6d3f-499e-82a0-d55133a4412a)
</details>

**Comment by josh-morgan117** [comment]
We need to update our m3 profile first before we can proceed with this one.

**Comment by ShanaLMoore** [comment]
@kidon0011 We QA'd based off of the profile on staging: https://dc.utk-hyku-staging.notch8.cloud/profiles/31/export?locale=en

please download version 35

and upload this version to prod.

If you upload a diff profile than what is on staging, we'd need to re QA it.

**Comment by ShanaLMoore** [comment]
Pulled back because LaRita noticed a problem with pdf splitting. ref: https://assaydepot.slack.com/archives/C031E2P1RJM/p1690906434704899

**Comment by jillpe** [comment]
blocked by #545 (which is follow up to LaRita's pdf splitting)
